app/services/audit/event/payload.rb

```diff
@@ -0,0 +1,26 @@
+module Audit
+  module Event
+    class Payload
+      ##
+      # Class where the Event payload is defined.
+      # Here are all the custom attributes we want to use in the event payload.
+      # Use attr_writer and Dry::Initializer.param method to define additional attributes.
+      #
+      # All custom attributes must define default values. This is due to Dry::Initializer,
+      # which isn't able to define new instance objects without attributes values.
+      # New instance object is used as block parameter in Audit::Notification.notify method.
+
+      extend Dry::Initializer
```

Comment:
I'm wondering if we really need this gem, for two reasons:

- it's yet another DSL which adds cognitive complexity to the reader
- it can be easily replaced by a constructor

I know it's supposed to cleanup code but adding production gems to an already big code base should have a good reason behind it and I'm not sure this is enough of a reason.

Isn't this more or less equivalent to:

```ruby
class Payload
  attr_accessor :user_id, :roles

  def initialize(data = {})
    @user_id = data[:user_id]
    @roles = data[:roles] || []
  end

  def to_s
    "user_id:#{user_id}, roles:#{roles}"
  end

  class << self
    def empty
      Payload.new
    end
  end
end

puts Payload::empty
puts Payload.new(user_id: "abc1234", roles: [:admin])
```
(the `to_s` is added just for demonstration purposes)

which outputs:

```ruby
user_id:, roles:[]
user_id:abc1234, roles:[:admin]
```

Reply:
@rhymes I think you are right. For this simple implementation, I don't need additional libraries.